### ISSUE TITLE
Provide a way to start reconnect logic for a broker that is unavailable at initial connect time

### ISSUE THREAD

**Issue by taladar** [stevedonovan/mosquitto-client#3]
Currently it is possible to connect to a broker and the client reconnects if the broker becomes unavailable temporarily after the initial connection.

It would be useful to be able to start this same reconnect logic when initially connecting too so the client can be restarted at any time and will just connect as soon as the broker becomes available.

**Comment by stevedonovan** [comment]
Good idea, although there we will be going beyond Mosquitto functionality. When wanting reliable connections I usually send out a 'heartbeat' message every minute or so, for reconnection to happen (long-lived TCP connections get stale).
